pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -115,7 +119,83 @@ lazy_static! {
 }
 
 lazy_static! {
-    static ref SYNC_QUEUE: SegQueue<SyncTask> = SegQueue::new();
+    static ref SYNC_QUEUE: SyncQueue = SyncQueue::new();
+}
+
+struct SyncQueue {
+    sender: UnboundedSender<SyncTask>,
+    receiver: Mutex<UnboundedReceiver<SyncTask>>,
+    length: AtomicUsize,
+}
```

Comment:
Consider using once_cell for the sender, and owning the receiver in the sync thread. Then you don't need a Mutex

Reply:
I like the idea, thanks. Had implemented it by moving away all queue-related things into a separate module.